utils/forest_shim/src/sector.rs

```diff
@@ -0,0 +1,102 @@
+        &self.0
+    }
+}
+
+impl From<RegisteredSealProofV2> for RegisteredSealProof {
+    fn from(value: RegisteredSealProofV2) -> RegisteredSealProof {
+        let proof = match value {
+            RegisteredSealProofV2::StackedDRG2KiBV1 => RegisteredSealProofV3::StackedDRG2KiBV1,
+            RegisteredSealProofV2::StackedDRG512MiBV1 => RegisteredSealProofV3::StackedDRG512MiBV1,
+            RegisteredSealProofV2::StackedDRG8MiBV1 => RegisteredSealProofV3::StackedDRG8MiBV1,
+            RegisteredSealProofV2::StackedDRG32GiBV1 => RegisteredSealProofV3::StackedDRG32GiBV1,
+            RegisteredSealProofV2::StackedDRG64GiBV1 => RegisteredSealProofV3::StackedDRG64GiBV1,
+            RegisteredSealProofV2::StackedDRG2KiBV1P1 => RegisteredSealProofV3::StackedDRG2KiBV1P1,
+            RegisteredSealProofV2::StackedDRG512MiBV1P1 => {
+                RegisteredSealProofV3::StackedDRG512MiBV1P1
+            }
+            RegisteredSealProofV2::StackedDRG8MiBV1P1 => RegisteredSealProofV3::StackedDRG8MiBV1P1,
+            RegisteredSealProofV2::StackedDRG32GiBV1P1 => {
+                RegisteredSealProofV3::StackedDRG32GiBV1P1
+            }
+            RegisteredSealProofV2::StackedDRG64GiBV1P1 => {
+                RegisteredSealProofV3::StackedDRG64GiBV1P1
+            }
+            RegisteredSealProofV2::Invalid(i64) => RegisteredSealProofV3::Invalid(i64),
+        };
+        RegisteredSealProof(proof)
+    }
+}
+
+impl From<RegisteredSealProof> for RegisteredSealProofV2 {
+    fn from(value: RegisteredSealProof) -> RegisteredSealProofV2 {
+        match value.0 {
+            RegisteredSealProofV3::StackedDRG2KiBV1 => RegisteredSealProofV2::StackedDRG2KiBV1,
+            RegisteredSealProofV3::StackedDRG512MiBV1 => RegisteredSealProofV2::StackedDRG512MiBV1,
+            RegisteredSealProofV3::StackedDRG8MiBV1 => RegisteredSealProofV2::StackedDRG8MiBV1,
+            RegisteredSealProofV3::StackedDRG32GiBV1 => RegisteredSealProofV2::StackedDRG32GiBV1,
+            RegisteredSealProofV3::StackedDRG64GiBV1 => RegisteredSealProofV2::StackedDRG64GiBV1,
+            RegisteredSealProofV3::StackedDRG2KiBV1P1 => RegisteredSealProofV2::StackedDRG2KiBV1P1,
+            RegisteredSealProofV3::StackedDRG512MiBV1P1 => {
+                RegisteredSealProofV2::StackedDRG512MiBV1P1
+            }
+            RegisteredSealProofV3::StackedDRG8MiBV1P1 => RegisteredSealProofV2::StackedDRG8MiBV1P1,
+            RegisteredSealProofV3::StackedDRG32GiBV1P1 => {
+                RegisteredSealProofV2::StackedDRG32GiBV1P1
+            }
+            RegisteredSealProofV3::StackedDRG64GiBV1P1 => {
+                RegisteredSealProofV2::StackedDRG64GiBV1P1
+            }
+            RegisteredSealProofV3::Invalid(i64) => RegisteredSealProofV2::Invalid(i64),
+        }
+    }
+}
+
+/// Represents a shim over SectorInfo from fvm_shared with convenience methods to convert to an older version of the type
+pub struct SectorInfo(pub SectorInfoV3);
```

Comment:
Same as for `RegisteredSealProof`, it would be nice to keep it encapsulated.

Reply:
done, thanks